controllers/useraccount/useraccount_controller_test.go

```diff
@@ -477,6 +478,159 @@ func TestReconcile(t *testing.T) {
 		})
 	})
 
+	t.Run("delete useraccount with console resources removes subsequent resources", func(t *testing.T) {
+		// when the console resources can only be found by name of the type "user-settings-UserUID"
+		cfg := commonconfig.NewMemberOperatorConfigWithReset(t)
+
+		mockCallsCounter := new(int)
+		userAcc := newUserAccount(username, userID)
+		util.AddFinalizer(userAcc, toolchainv1alpha1.FinalizerName)
+		resourceName := ConsoleUserSettingsResourceNamePrefix + string(userUID)
+		configMap := &corev1.ConfigMap{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      resourceName,
+				Namespace: UserSettingNS,
+			},
+		}
+		role := &v1.Role{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      resourceName + "random",
+				Namespace: UserSettingNS,
+				Labels: map[string]string{
+					ConsoleUserSettingsIdentifier: "true",
+					ConsoleUserSettingsUID:        string(userUID),
+				},
+			},
+		}
+		rb := &v1.RoleBinding{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      resourceName,
+				Namespace: UserSettingNS,
+			},
+		}
+		noiseCm := &corev1.ConfigMap{
+			TypeMeta: metav1.TypeMeta{
+				Kind:       "ConfigMap",
+				APIVersion: "v1",
+			},
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      "noise-object",
+				Namespace: UserSettingNS,
+			},
+		}
+		noiseRole := &v1.Role{
+			TypeMeta: metav1.TypeMeta{
+				Kind:       "Role",
+				APIVersion: "rbac.authorization.k8s.io/v1",
+			},
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      resourceName + "noise-role",
+				Namespace: UserSettingNS,
+				Labels: map[string]string{
+					ConsoleUserSettingsIdentifier: "true",
+				},
+			},
+		}
+		noiseRb := &v1.RoleBinding{
+			TypeMeta: metav1.TypeMeta{
+				Kind:       "RoleBinding",
+				APIVersion: "rbac.authorization.k8s.io/v1",
+			},
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      "noise-rb",
+				Namespace: UserSettingNS,
+			},
+		}
+		r, req, cl, _ := prepareReconcile(t, username, cfg, userAcc, preexistingUser, preexistingIdentity, configMap, role, rb, noiseCm, noiseRole, noiseRb)
+
+		t.Run("first reconcile deletes identity", func(t *testing.T) {
+			// given
+			// Set the deletionTimestamp
+			now := metav1.NewTime(time.Now())
+			userAcc.DeletionTimestamp = &now
+			err = r.Client.Update(context.TODO(), userAcc)
+			require.NoError(t, err)
+
+			// when
+			res, err := r.Reconcile(context.TODO(), req)
+
+			// then
+			assert.Equal(t, reconcile.Result{}, res)
+			require.NoError(t, err)
+
+			useraccount.AssertThatUserAccount(t, req.Name, cl).
+				HasConditions(notReady("Terminating", "deleting user/identity"))
+
+			// Check that the associated identity has been deleted
+			// when reconciling the useraccount with a deletion timestamp
+			assertIdentityNotFound(t, r, userAcc, config.Auth().Idp())
+			useraccount.AssertThatUserAccount(t, userAcc.Name, cl).HasConditions(terminating())
```

Comment:
this seems to be a duplication of what is already tested in the previous test. Couldn't this be merged or refactored?

Reply:
done in [7b7bd25](https://github.com/codeready-toolchain/member-operator/pull/568/commits/7b7bd259bb5cb92d555a5043585a0d3c06875e8b)